src/main/java/org/springframework/integration/aws/outbound/KinesisMessageHandler.java

```diff
@@ -163,19 +211,21 @@ protected void onInit() throws Exception {
 
 	@Override
 	@SuppressWarnings("unchecked")
-	protected void handleMessageInternal(Message<?> message) throws Exception {
+	protected void handleMessageInternal(final Message<?> message) throws Exception {
 		Future<?> resultFuture = null;
+
 		if (message.getPayload() instanceof PutRecordsRequest) {
+
 			resultFuture = this.amazonKinesis.putRecordsAsync((PutRecordsRequest) message.getPayload(),
-					(AsyncHandler<PutRecordsRequest, PutRecordsResult>) this.asyncHandler);
+					getPutRecordsAsyncHandler(message));
```

Comment:
We don't need this method. We simply can call the second one: `getPutRecordAsyncHandler(message, (PutRecordsRequest) message.getPayload())`